plugins/out_s3/s3.c

```diff
@@ -1243,8 +1308,18 @@ static int put_all_chunks(struct flb_s3 *ctx)
                                 chunk->create_time, buffer, buffer_size);
             flb_free(buffer);
             if (ret < 0) {
-                s3_store_file_unlock(chunk);
                 chunk->failures += 1;
+                if (chunk->failures > ctx->ins->retry_limit){
+                    less_than_limit = FLB_FALSE;
```

Comment:
I think `less_than_limit` isn't strictly needed here since the if/else breaks down the two cases, and so you could just directly pass FLB_TRUE or FLB_FALSE to s3_retry_warn

Reply:
I think you could also use this pattern everywhere that you have to check if you are less than or over the limit...